### PR TITLE
Replace samples with kustomize examples

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

**Review by slagle — COMMENTED** [review]
I like the direction.
Instead of examples/base/openstackdataplanenodeset.yaml, can just make that use config/samples/dataplane_v1beta1_openstackdataplanenodeset.yaml? 

That would further remove duplication between the values.yaml and the sample.

I'd also like to get all fields that are just default values removed from the samples, so that they are minimal. We could have a single sample for each resource that documents all defaults as well.

**Comment by fao89** [comment]
> I like the direction. Instead of examples/base/openstackdataplanenodeset.yaml, can just make that use config/samples/dataplane_v1beta1_openstackdataplanenodeset.yaml?
> 

there is some customize logic for building the bundle, so I can't use `config/samples/dataplane_v1beta1_openstackdataplanenodeset.yaml`, what I can do is copy its content to `examples/base/openstackdataplanenodeset.yaml`

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

File: examples/baremetal/values.yaml
Comment (rabi) [inline examples/baremetal/values.yaml:20]: Would not this create more issues and confusion as values.yaml has networks as a subkey in nodeset where as it maps to spec.nodeTemplate.networks?

File: examples/bgp/values.yaml
Comment (rabi) [inline examples/bgp/values.yaml:87]: Do we need to repeat the `nodes` section as they're in the base? only the delta can be kustomized I assume. Also, we're now going to maintain 3 files for a sample and then the dependency on the base.

**Review by jpodivin — CHANGES_REQUESTED** [review]
I think documentation should be part of this PR. They target the same repo and make sense only together.

Plus if anything changes here you'll have to remember to make changes there.

**Comment by fao89** [comment]
> I think documentation should be part of this PR. They target the same repo and make sense only together.
> 
> Plus if anything changes here you'll have to remember to make changes there.

I don't want to make this PR bigger, I'll keep the other PR on hold until we merge this one

**Comment by fao89** [comment]
recheck

**Review by fultonj — COMMENTED** [review]
I am in favor of this proposed change.

For someone who only wants to refernce samples things change a little. Instead of looking directly at the samples [1] they can clone the repo and run kustomize and see the same thing. E.g. I can run `kustomize build examples/pre_ceph_hci/` and see the same outputs I would see if I clicked on [2] and [3].

This makes the samples much more maintainable as all the different examples can inherit from the base and duplicate data is replaced by references. 

I also see that for any change  in the new examples diretory, that you make sure kustomize can build it [4].

Thanks @fao89 

[1] https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/config/samples

[2] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplanedeployment_pre_ceph_hci.yaml

[3] https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplanedeployment_pre_ceph_hci.yaml

[4] https://github.com/openstack-k8s-operators/dataplane-operator/actions/runs/8085585345/workflow?pr=698#L30-L33

**Review by slagle — COMMENTED** [review]
LGTM, +2
How do we intend to merge this? Just merge this PR and the install_yamls PR as close together as possible?

**Comment by fao89** [comment]
> LGTM, +2 How do we intend to merge this? Just merge this PR and the install_yamls PR as close together as possible?

I'll drop the breaking change on this PR, so it will be easier to coordinate with install_yamls

**Comment by fao89** [comment]
recheck

**Comment by slagle** [comment]
recheck

https://github.com/openstack-k8s-operators/install_yamls/pull/752

**Review by slagle — COMMENTED** [review]
Can you restore:
config/samples/dataplane_v1beta1_openstackdataplanenodeset_bgp.yaml
config/samples/dataplane_v1beta1_openstackdataplanenodeset_bgp_ovn_cluster.yaml
config/samples/dataplane_v1beta1_openstackdataplanenodeset_networker.yaml
config/samples/dataplane_v1beta1_openstackdataplanedeployment_networker.yaml

that way we can merge this without breaking any part of install_yamls?

I think bumping the default KUSTOMIZE_VERSION to v5.0.1 in the Makefile is required as well, since this won't work with v3.8.7

**Comment by fao89** [comment]
I kept the samples that were needed on CI, the other samples is more for development and I believe it won't take long for https://github.com/openstack-k8s-operators/install_yamls/pull/743 to get merged

**Review by slagle — APPROVED** [review]

**Pull request merged**
